WeakAurasOptions/WeakAurasOptions.lua

```diff
@@ -892,38 +890,102 @@ function WeakAuras.SortDisplayButtons(filter, overrideReset, id)
 
   wipe(frame.buttonsScroll.children);
 
-  local pendingImportButtonShown = false
+  local pendingInstallButtonShown = false
   local CompanionData = WeakAurasCompanion and WeakAurasCompanion.WeakAuras or WeakAurasCompanion
   if CompanionData and CompanionData.stash then
     for id, companionData in pairs(CompanionData.stash) do
-      if not pendingImportButtonShown then
-        tinsert(frame.buttonsScroll.children, frame.pendingImportButton)
-        pendingImportButtonShown = true
+      if not pendingInstallButtonShown then
+        tinsert(frame.buttonsScroll.children, frame.pendingInstallButton)
+        pendingInstallButtonShown = true
       end
-      if frame.pendingImportButton:GetExpanded() then
-        if not(pendingUpdateButtons[id]) then
-          pendingUpdateButtons[id] = AceGUI:Create("WeakAurasPendingUpdateButton")
-          pendingUpdateButtons[id]:Initialize(id, companionData)
+      local child = pendingInstallButtons[id]
+      if frame.pendingInstallButton:GetExpanded() then
+        if not child then
+          child = AceGUI:Create("WeakAurasPendingInstallButton")
+          pendingInstallButtons[id] = child
+          child:Initialize(id, companionData)
           if companionData.logo then
-            pendingUpdateButtons[id]:SetLogo(companionData.logo)
+            child:SetLogo(companionData.logo)
           end
           if companionData.refreshLogo then
-            pendingUpdateButtons[id]:SetRefreshLogo(companionData.refreshLogo)
+            child:SetRefreshLogo(companionData.refreshLogo)
           end
         end
-        pendingUpdateButtons[id].frame:Show()
-        pendingUpdateButtons[id]:AcquireThumbnail()
-        frame.buttonsScroll:AddChild(pendingUpdateButtons[id])
-      elseif pendingUpdateButtons[id] then
-        pendingUpdateButtons[id].frame:Hide()
-        if pendingUpdateButtons[id].ReleaseThumbnail then
-          pendingUpdateButtons[id]:ReleaseThumbnail()
+        child.frame:Show()
+        child:AcquireThumbnail()
+        frame.buttonsScroll:AddChild(child)
+      elseif child then
+        child.frame:Hide()
+        if child.ReleaseThumbnail then
+          child:ReleaseThumbnail()
         end
       end
     end
   end
-  if not pendingImportButtonShown and frame.pendingImportButton then
-    frame.pendingImportButton.frame:Hide()
+  if not pendingInstallButtonShown and frame.pendingInstallButton then
+    frame.pendingInstallButton.frame:Hide()
+  end
+
+  local pendingUpdateButtonShown = false
+  if CompanionData then
+    local handled = {}
+    for id, aura in pairs(WeakAurasSaved.displays) do
```

Comment:
Doing this just on initialization does miss two things:

New auras that are imported that match a existing pending import. (Frankly not that likely)
Auras being deleted which match a pending import. (I haven't tested what happens in that case.)